### PR TITLE
Add opencus and edit delcus

### PR DESCRIPTION
**Pull request by nvmdava**
Also edited data to be more readable. Later need to discuss how to make the pane auto switch to commissions.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/OpenCustomerCommandParser.java [hunk 1]
```diff
@@ -3,26 +3,26 @@
...
 /**
- * Parses input arguments and creates a new DeleteCommand object
+ * Parses input arguments and creates a new OpenCustomerCommand object
  */
-public class DeleteCommandParser implements Parser<DeleteCommand> {
+public class OpenCustomerCommandParser implements Parser<OpenCustomerCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteCommand
      * and returns a DeleteCommand object for execution.
```
Comment (seelengxd) [inline src/main/java/seedu/address/logic/parser/OpenCustomerCommandParser.java:16]:
```suggestion
     * Parses the given {@code String} of arguments in the context of the OpenCustomerCommand
     * and returns a OpenCustomerCommand object for execution.
 ```

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 2]
```diff
@@ -58,6 +59,10 @@ public class MainWindow extends UiPart<Stage> {
     @FXML
     private StackPane statusbarPlaceholder;
 
+    @FXML
+    private TabPane tabPane;
+
+
```
Comment (seelengxd) [inline src/main/java/seedu/address/ui/MainWindow.java:65]:
```suggestion
```
Should we remove the extra new line?

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 3]
```diff
@@ -58,6 +59,10 @@ public class MainWindow extends UiPart<Stage> {
     @FXML
     private StackPane statusbarPlaceholder;
 
+    @FXML
+    private TabPane tabPane;
+
```
Comment (seelengxd) [inline src/main/java/seedu/address/ui/MainWindow.java:64]: This variable is currently unused in this PR, but I am assuming this is to be used for switching panes in the future - along with the change to `MainWindow.fxml`.

Helpful link for future reference: https://stackoverflow.com/questions/6902377/javafx-tabpane-how-to-set-the-selected-tab

File: src/main/java/seedu/address/logic/commands/OpenCustomerCommand.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        List<Customer> lastShownList = model.getFilteredCustomerList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_CUSTOMER_DISPLAYED_INDEX);
+        }
+
+        Customer customerToOpen = lastShownList.get(targetIndex.getZeroBased());
+        model.selectCustomer(customerToOpen);
+
+        model.updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);
```
Comment (seelengxd) [inline src/main/java/seedu/address/logic/commands/OpenCustomerCommand.java:47]:
```suggestion
```

This line currently would reset the search if the user does (2nd user appears w/o list). Not sure if this is a bug or a feature.
```
find Trixx
opencom 1
```

File: src/main/java/seedu/address/logic/commands/OpenCustomerCommand.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.customer.Customer;
+
+/**
+ * Deletes a and shows customer details and switches tab to commissions to show commissions made by the customer.
```
Comment (seelengxd) [inline src/main/java/seedu/address/logic/commands/OpenCustomerCommand.java:15]:
```suggestion
 * Selects a customer and shows their details.
```

File: src/test/java/seedu/address/logic/commands/OpenCustomerCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    public void equals() {
+        OpenCustomerCommand openFirstCommand = new OpenCustomerCommand(INDEX_FIRST_CUSTOMER);
+        OpenCustomerCommand openSecondCommand = new OpenCustomerCommand(INDEX_SECOND_CUSTOMER);
+
+        // same object -> returns true
+        assertTrue(openFirstCommand.equals(openFirstCommand));
+
+        // same values -> returns true
+        OpenCustomerCommand deleteFirstCommandCopy = new OpenCustomerCommand(INDEX_FIRST_CUSTOMER);
+        assertTrue(openFirstCommand.equals(deleteFirstCommandCopy));
```
Comment (seelengxd) [inline src/test/java/seedu/address/logic/commands/OpenCustomerCommandTest.java:86]:
```suggestion
        OpenCustomerCommand openFirstCommandCopy = new OpenCustomerCommand(INDEX_FIRST_CUSTOMER);
        assertTrue(openFirstCommand.equals(openFirstCommandCopy));
```

**Review by seelengxd — CHANGES_REQUESTED** [review]
Minor issues to fix.

**Review by ngxingyu — COMMENTED** [review]
Looks fine. Just 2 points
1. For the case of `delcus`, I would prefer the entire list of all customers to be shown (`updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);`) called somewhere, otherwise we will get an empty list after running the following:
```
find Trixx
del 1
```
2. when we call `opencus`, it would be good to get the UI to focus on the active customer in the ListView since clicking on customer X will focus on it, calling opencus on another customer will change the active customer but leave customer X focused. We can open an issue for this unless it's something easy for you to fix

**Comment by nvmdava** [comment]
Because our logic will need to call commands on the UI directly(same for switching tabs), I think we should discuss how we want to do that. After the pull request is merged I will add issues for it. Thanks, good eye on find and deleting.

**Review by seelengxd — APPROVED** [review]
LGTM!

**Pull request merged**